packages/core-components/src/components/AutoLogoutProvider/AutoLogoutProvider.tsx

```diff
@@ -0,0 +1,211 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { identityApiRef, useApi } from '@backstage/core-plugin-api';
+import React, { PropsWithChildren, useEffect, useMemo, useState } from 'react';
+import {
+  EventsType,
+  IdleTimerProvider,
+  IIdleTimer,
+  workerTimers,
+} from 'react-idle-timer';
+
+import {
+  LAST_SEEN_ONLINE_STORAGE_KEY,
+  useLogoutDisconnectedUserEffect,
+} from './disconnectedUsers';
+import { StillTherePrompt } from './StillTherePrompt';
+import { DefaultTimestampStore, TimestampStore } from './timestampStore';
+
+/** @public */
+export type AutoLogoutTrackableEvent = EventsType;
+
+/** @public */
+export type AutoLogoutProviderProps = {
+  /**
+   * Enable/disable the AutoLogoutMechanism.
+   * defaults to true
+   */
+  enabled?: boolean;
+  /**
+   * The amount of time (in minutes) of inactivity
+   * after which the user is automatically logged out.
+   * defaults to 60 minutes.
+   */
+  idleTimeoutMinutes?: number;
+  /**
+   * The number of seconds before the idleTimeout expires,
+   * at which the user will be alerted by a Dialog that
+   * they are about to be logged out.
+   * defaults to 10 seconds
+   */
+  promptBeforeIdleSeconds?: number;
+  /**
+   * Enable/disable the usage of Node's worker thread timers instead of main thread timers.
+   * This is helpful if you notice that the your browser is killing inactive tab's timers, like the one used by AutoLogoutProvider.
+   * If you experience some browser incompatibility, you may try to set this to false.
+   * defaults to true.
+   */
+  useWorkerTimers?: boolean;
+  /**
+   * List of DOM events that the AutoLogoutProvider will track to determine if the user is active or not.
+   * default list includes all the needed events for keyboard/pointers/mouse devices.
+   */
+  events?: AutoLogoutTrackableEvent[];
+  /**
+   * Enable/disable the autologout for disconnected users.
+   * disconnected users are the ones that are logged in but have no Backstage tab open in their browsers.
+   * If enabled, disconnected users will be automatically logged out after `idleTimeoutMinutes`
+   * defaults to true
+   */
+  logoutIfDisconnected?: boolean;
+};
+
+/**
+ * The Autologout feature enables platform engineers to add a mechanism to log out users after a configurable amount of time of inactivity.
+ * When enabled, the mechanism will track user actions (mouse movement, mouse click, key pressing, taps, etc.) in order to determine if they are active or not.
+ * After a certain amount of inactivity/idle time, the user session is invalidated and they are required to sign in again.
+ *
+ * @public
+ */
+export const AutoLogoutProvider = ({
```

Comment:
A reason this is a provider is to avoid the conditional hook call you will have with `useIdleTimer`. I want the timer to be fully disabled/non-existent when user sets `enabled = false` (even through app-config). any suggestion on this? 
Also, the `<StillTherePrompt>` is using the context, but in this case I can easily pass a reference to the actual timer.